app/src/main/java/org/oppia/android/app/devoptions/testing/DeveloperOptionsTestActivity.kt

```diff
@@ -79,12 +85,13 @@ class DeveloperOptionsTestActivity :
   }
 
   companion object {
-    const val PROFILE_ID_EXTRA_KEY = "DeveloperOptionsTestActivity.profile_id"
-
     /** Returns [Intent] for [DeveloperOptionsTestActivity]. */
     fun createDeveloperOptionsTestIntent(context: Context, internalProfileId: Int): Intent {
+      val profileId = ProfileId.newBuilder().setInternalId(internalProfileId).build()
       val intent = Intent(context, DeveloperOptionsActivity::class.java)
-      intent.putExtra(PROFILE_ID_EXTRA_KEY, internalProfileId)
+        .apply {
+          decorateWithUserProfileId(profileId)
+        }
```

Comment:
Can be flattened to:
```suggestion
      intent.decorateWithUserProfileId(profileId)
```

Reply:
Done.